packages/components/psammead-episode-list/src/episode.jsx

```diff
@@ -0,0 +1,71 @@
+import React from 'react';
+import styled from 'styled-components';
+import { C_METAL, C_POSTBOX } from '@bbc/psammead-styles/colours';
+
+import { string, shape, func, arrayOf, element } from 'prop-types';
+import { scriptPropType } from '@bbc/gel-foundations/prop-types';
+
+import PlayButton from './playButton';
+
+const getEpisodeWrapper = ElementType => styled(ElementType)`
+  a {
+    display: block;
+    &:hover,
+    &:focus {
+      span {
+        text-decoration: underline;
+      }
+      .play-button-wrapper {
+        background-color: ${C_POSTBOX};
+        border-color: ${C_POSTBOX};
+      }
+      .play-icon {
+        border-left-color: white;
+      }
+    }
+    &:visited {
+      span {
+        color: ${C_METAL};
+      }
+    }
+  }
+`;
+
+const Wrapper = styled.div`
+  display: inline-block;
+`;
+
+const Episode = ({ children, as, LinkElement, script, service }) => {
+  const EpisodeWrapper = getEpisodeWrapper(as);
+  return (
+    <EpisodeWrapper>
+      <LinkElement>
+        <PlayButton />
+        <Wrapper>
+          {children.map(child =>
+            React.cloneElement(child, {
+              key: child.props.children,
```

Comment:
What is the purpose of the cloning here? Also, is it correct to set a key as the children?

Reply:
👍 fixed by refactor to use context to pass props to children